### PR TITLE
fix(caldav): Ignore invalid events for reminder generation

### PR DESCRIPTION
**Pull request by ChristophWurst**
## Summary

Fixes ``Call to a member function serialize() on null at apps/dav/lib/CalDAV/Reminder/ReminderService.php:261`` with 25.0.4.

## TODO

- [x] Fix

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

File: apps/dav/lib/CalDAV/Reminder/ReminderService.php [hunk 1]
```diff
@@ -267,7 +275,14 @@ public function onCalendarObjectCreate(array $objectData):void {
 		if ($masterItem) {
 			$processedAlarms = [];
 			$masterAlarms = [];
-			$masterHash = $this->getEventHash($masterItem);
+			try {
+				$masterHash = $this->getEventHash($recurrenceException);
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Reminder/ReminderService.php:279]: **nitpick:** 
```suggestion
				$masterHash = $this->getEventHash($masterItem);
```

File: apps/dav/lib/CalDAV/Reminder/ReminderService.php [hunk 2]
```diff
@@ -243,7 +244,14 @@ public function onCalendarObjectCreate(array $objectData):void {
 		$isRecurring = $masterItem ? $this->isRecurring($masterItem) : false;
 
 		foreach ($recurrenceExceptions as $recurrenceException) {
-			$eventHash = $this->getEventHash($recurrenceException);
+			try {
+				$eventHash = $this->getEventHash($recurrenceException);
+			} catch (Throwable $e) {
```
Comment (tcitworld) [inline apps/dav/lib/CalDAV/Reminder/ReminderService.php:249]: **question:** Which exceptions do you except to be sent here? From what I can see serializing properties, using `implode` and `md5` shouldn't throw anything.

Comment (ChristophWurst) [reply apps/dav/lib/CalDAV/Reminder/ReminderService.php:249]: ``Call to a member function serialize() on null at apps/dav/lib/CalDAV/Reminder/ReminderService.php:261``

Added it to the PR description too. We saw this on a production instance, during the upgrade from an old version. Apparently there can be events without a DTSTART.

Comment (ChristophWurst) [reply apps/dav/lib/CalDAV/Reminder/ReminderService.php:249]: Do you have any ideas for a better place to catch this? We might be able to skip invalid events all together for the reminder processing.

Comment (tcitworld) [reply apps/dav/lib/CalDAV/Reminder/ReminderService.php:249]: Not catching, but we can simply validate events in `getAllVEventsFromVCalendar`.

Something like:
```php
if (!$child->DTSTART) {
  continue
}
```

File: apps/dav/tests/unit/CalDAV/Reminder/ReminderServiceTest.php [hunk 3]
```diff
@@ -342,6 +342,53 @@ public function testOnCalendarObjectCreateSingleEntry():void {
...
+END:VCALENDAR
+EOD;
+		$objectData = [
+			'calendardata' => $calendarData,
+			'id' => '42',
+			'calendarid' => '1337',
+			'component' => 'vevent',
+		];
+
+		$this->backend->expects($this->exactly(2))
```
Comment (tcitworld) [inline apps/dav/tests/unit/CalDAV/Reminder/ReminderServiceTest.php:376]: **question:** Shouldn't the method never be called?

**Review by tcitworld — CHANGES_REQUESTED** [review]

**Review by tcitworld — APPROVED** [review]

**Review by kesselb — APPROVED** [review]

**Review by hamza221 — APPROVED** [review]

**Comment by ChristophWurst** [comment]
the backport worked locally

**Pull request merged**